### ISSUE TITLE
Anyone else getting slow download speed from youtube? 

### ISSUE THREAD

**Issue by peter099** [ytdl-org/youtube-dl#30354]
## Checklist

- [ x] I'm asking a question
- [x ] I've looked through the README and FAQ for similar questions
- [x ] I've searched the bugtracker for similar questions including closed ones

## Question

My download speed from youtube is extremely slow - it doesn't go more than 100 KiB/s. On other sites, like Rumble for example, download speed is fine. Anyone else having this issue?

**Comment by FifiTheBulldog** [comment]
There are already a ton of issues, both open and closed, that have been posted here regarding slow download speeds.

**Comment by dirkf** [comment]
How true.

Duplicate of #29326, fixed in PR #30184.

**Comment by fastily** [comment]
This issue was fixed months ago in the (actively maintained) spiritual successor, [yt-dlp](https://github.com/yt-dlp/yt-dlp)

**Comment by peter099** [comment]
> This issue was fixed months ago in the (actively maintained) spiritual successor, [yt-dlp](https://github.com/yt-dlp/yt-dlp)

Do you mean youtube-dl will not get any more updates?

**Comment by ksharindam** [comment]
I am also getting very slow download speed. Around 50KB/s.
Facing this issue since last few months. Have not yet been fixed.

**Comment by EsterLoken** [comment]
> > This issue was fixed months ago in the (actively maintained) spiritual successor, [yt-dlp](https://github.com/yt-dlp/yt-dlp)
> 
> Do you mean youtube-dl will not get any more updates?

Who knows, I wouldn't hold my breath.. Looking at the last couple months it doesn't make sense to stick with youtube-dl at this point, yt-dlp has superseded it in any way and if youtube-dl ever picks up development again, they'll incorporate any useful changes anyway.

yt-dlp is fully backwards compatible with youtube-dl and under active development with several releases per month, [full of bugfixes and requested features.](https://github.com/yt-dlp/yt-dlp/releases)

Meanwhile youtube-dl had its last release in _June_. And the last time [anyone touched any part of youtube-dl was July 1st.](https://github.com/ytdl-org/youtube-dl/commits/master) I think it's pretty safe to say ...

![image](https://user-images.githubusercontent.com/33354862/145729078-d3d9f907-cb87-46ab-9bbd-af9552056ffa.png)

**Comment by Kirmakoff** [comment]
Thank you for recommending yt-dlp. 
It's not easy to find a good successor among thousands of forks for youtube-dl.
I confirm that it circumvents the bug and gives decent download speed, along with fancy styling:

![изображение](https://user-images.githubusercontent.com/15086622/145889495-5913356f-1765-47a5-9c6b-2b0a813b5e8c.png)

**Comment by 89z** [comment]
I have an alternate tool that works:

https://github.com/89z/mech

**Comment by andyncake** [comment]
> 

I do think your project is a scam or spam. 

You have <= 100 followers and your project does nothing in terms of video fetching.

**Comment by 89z** [comment]
@andyncake you are right, its totally a scam:

~~~
PS C:\> youtube ABkF3MSxoQg
POST https://www.youtube.com/youtubei/v1/player
GET https://rr4---sn-q4flrnle.googlevideo.com/videoplayback?expire=1639863071&ei=v_69YZCrHaCAir4PiYOk2AY&ip=72.181.23.38&id=o-ABqtO3SJ0We57cF4QN20wmJyhS-wuqdGjbS99QB1sX5X&itag=247&source=youtube&requiressl=yes&mh=nP&mm=31%2C26&mn=sn-q4flrnle%2Csn-5uaeznyz&ms=au%2Conr&mv=m&mvi=4&pl=15&initcwndbps=2046250&vprv=1&mime=video%2Fwebm&gir=yes&clen=16329486&dur=302.468&lmt=1611116994001489&mt=1639841088&fvip=4&keepalive=yes&fexp=24001373%2C24007246&c=ANDROID&txp=5535432&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIhAJ7rOW1X8DvJauJ7PGLWrFvtgh614KiyZD5uMuwOj30qAiBraAxmjOxdOm7Lp2MMPSvlJTs4LQMXyW_uEH3esT6ubA%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRAIgb16THGTWYVti2V2qlTIWO8kvNvEs3aTlJ-nUqxuy4soCIEosQwhIl2IddS7dXlozVMgustOQXL_RoANqMaZ9wjaF
0% bytes=0-9999999
61% bytes=10000000-19999999 8.208 MB/s
GET https://rr4---sn-q4flrnle.googlevideo.com/videoplayback?expire=1639863071&ei=v_69YZCrHaCAir4PiYOk2AY&ip=72.181.23.38&id=o-ABqtO3SJ0We57cF4QN20wmJyhS-wuqdGjbS99QB1sX5X&itag=251&source=youtube&requiressl=yes&mh=nP&mm=31%2C26&mn=sn-q4flrnle%2Csn-5uaeznyz&ms=au%2Conr&mv=m&mvi=4&pl=15&initcwndbps=2046250&vprv=1&mime=audio%2Fwebm&gir=yes&clen=4369482&dur=302.501&lmt=1611115659214938&mt=1639841088&fvip=4&keepalive=yes&fexp=24001373%2C24007246&c=ANDROID&txp=5531432&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRgIhAO-C4iXQ2Y8RMrdSFb7yJiUekwjoV72bV3fDE7lj6EJhAiEAzEIJf0p6Fx3lPAGJfkuH1ukrQ3zwcVPNC7qIhSI_I-I%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRAIgb16THGTWYVti2V2qlTIWO8kvNvEs3aTlJ-nUqxuy4soCIEosQwhIl2IddS7dXlozVMgustOQXL_RoANqMaZ9wjaF
0% bytes=0-9999999
~~~

**Comment by andyncake** [comment]
> @andyncake you are right, its totally a scam:
> 
> ```
> PS C:\> youtube ABkF3MSxoQg
> POST https://www.youtube.com/youtubei/v1/player
> GET https://rr4---sn-q4flrnle.googlevideo.com/videoplayback?expire=1639863071&ei=v_69YZCrHaCAir4PiYOk2AY&ip=72.181.23.38&id=o-ABqtO3SJ0We57cF4QN20wmJyhS-wuqdGjbS99QB1sX5X&itag=247&source=youtube&requiressl=yes&mh=nP&mm=31%2C26&mn=sn-q4flrnle%2Csn-5uaeznyz&ms=au%2Conr&mv=m&mvi=4&pl=15&initcwndbps=2046250&vprv=1&mime=video%2Fwebm&gir=yes&clen=16329486&dur=302.468&lmt=1611116994001489&mt=1639841088&fvip=4&keepalive=yes&fexp=24001373%2C24007246&c=ANDROID&txp=5535432&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIhAJ7rOW1X8DvJauJ7PGLWrFvtgh614KiyZD5uMuwOj30qAiBraAxmjOxdOm7Lp2MMPSvlJTs4LQMXyW_uEH3esT6ubA%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRAIgb16THGTWYVti2V2qlTIWO8kvNvEs3aTlJ-nUqxuy4soCIEosQwhIl2IddS7dXlozVMgustOQXL_RoANqMaZ9wjaF
> 0% bytes=0-9999999
> 61% bytes=10000000-19999999 8.208 MB/s
> GET https://rr4---sn-q4flrnle.googlevideo.com/videoplayback?expire=1639863071&ei=v_69YZCrHaCAir4PiYOk2AY&ip=72.181.23.38&id=o-ABqtO3SJ0We57cF4QN20wmJyhS-wuqdGjbS99QB1sX5X&itag=251&source=youtube&requiressl=yes&mh=nP&mm=31%2C26&mn=sn-q4flrnle%2Csn-5uaeznyz&ms=au%2Conr&mv=m&mvi=4&pl=15&initcwndbps=2046250&vprv=1&mime=audio%2Fwebm&gir=yes&clen=4369482&dur=302.501&lmt=1611115659214938&mt=1639841088&fvip=4&keepalive=yes&fexp=24001373%2C24007246&c=ANDROID&txp=5531432&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRgIhAO-C4iXQ2Y8RMrdSFb7yJiUekwjoV72bV3fDE7lj6EJhAiEAzEIJf0p6Fx3lPAGJfkuH1ukrQ3zwcVPNC7qIhSI_I-I%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRAIgb16THGTWYVti2V2qlTIWO8kvNvEs3aTlJ-nUqxuy4soCIEosQwhIl2IddS7dXlozVMgustOQXL_RoANqMaZ9wjaF
> 0% bytes=0-9999999
> ```
> 
> https://github.com/89z/mech

Where are the commands to download videos?

**Comment by 89z** [comment]
> Where are the commands to download videos?

https://github.com/89z/mech/releases/tag/v1.42.0

**Comment by andyncake** [comment]
> > Where are the commands to download videos?
> 
> https://github.com/89z/mech/releases/tag/v1.42.0

Oh come on, show me how to download this [https://www.youtube.com/watch?v=lugeruSbnAE](url)

**Comment by 89z** [comment]
@andyncake please stop spamming this issue. I am well known on this issue tracker, and while some people might not like my personality, then will confirm that I make a valid tool. If you have issue with my tool, then post on the correct tracker.

**Comment by andyncake** [comment]
> @andyncake please stop spamming this issue. I am well known on this issue tracker, and while some people might not like my personality, they will confirm that I make a valid tool. If you have issue with my tool, then post on the correct tracker.

Where's the user manual? 
How to use it? 
**Stop circumventing the core problems.** **Stop tricking people here**.

**Comment by 89z** [comment]
@andyncake you sir are a professional idiot. I already commented on the syntax, **which you even quoted**:

https://github.com/ytdl-org/youtube-dl/issues/30354#issuecomment-997227529

so please, just shut the hell up already.

**Comment by dragonfirefox** [comment]
> ## Checklist
> * [ x] I'm asking a question
> * [x ] I've looked through the README and FAQ for similar questions
> * [x ] I've searched the bugtracker for similar questions including closed ones
> 
> ## Question
> My download speed from youtube is extremely slow - it doesn't go more than 100 KiB/s. On other sites, like Rumble for example, download speed is fine. Anyone else having this issue?

I have same exact problem, you can use yt-dlp as alternative, it also available in python pkgs

**Comment by EsterLoken** [comment]
> I have same exact problem, you can use yt-dlp as alternative

I think at this point it's [not a case of can, but _should_.](https://github.com/ytdl-org/youtube-dl/commit/21b759057502c6e70d51011cfb3fb86d84055182)

**Comment by dirkf** [comment]
Closed in af9e72507.
